src/client/QXmppOutgoingClient.cpp

```diff
@@ -563,6 +563,8 @@ void QXmppOutgoingClient::handleStanza(const QDomElement &nodeRecv)
 
         if (!nodeRecv.firstChildElement("conflict").isNull())
             d->xmppStreamError = QXmppStanza::Error::Conflict;
+        else if (!nodeRecv.firstChildElement("not-authorized").isNull() || !nodeRecv.firstChildElement("bad-auth").isNull())
```

Comment:
The 'bad-auth' thing was a workaround for (old) prosody servers. Could you remove that. I think we can merge it then.

Reply:
Done. Fixed in b2bcfd9